### ISSUE TITLE
Dose the data division for multi-gpu destruct the accuracy?

### ISSUE THREAD

**Issue by venuswu** [tensorflow/models#7661]
Data division for multi-gpu after tf1.11 seems to have the same shape of data across all gpus, which will result in a bigger variance of loss between different steps. For example, for the transformer model, when in one step, all the gpus are feeded with the data with length near 8, but in the next step, all the gpus are feeded with the data with length near 32, which will result in a bigger variance of loss.

In practice, I got a lower accuracy trained with tf2.0 than tf1.11.

@guptapriya @yuefengz @zongweiz @sgpyc @saberkun

**Comment by saberkun** [comment]
Is this because you used dynamic batch size? Also, this is related to model tuning.

**Comment by venuswu** [comment]
> Is this because you used dynamic batch size? Also, this is related to model tuning.

Yes, I used dynamic batch size. @saberkun
